kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/KafkaPublisher.java

```diff
@@ -133,123 +121,77 @@ public void shutDown() {
      * available.
      * <p>
      * If {@link ProducerFactory} is configured to use:
+     * </p>
      * <ul>
      * <li>Transactions: use kafka transactions for publishing events</li>
      * <li>Ack: send messages and wait for acknowledgement from Kafka. Acknowledgement timeout can be configured via
      * {@link KafkaPublisher.Builder#publisherAckTimeout(long)}).</li>
      * <li>None: fire and forget.</li>
      * </ul>
      *
-     * @param events the events to publish on the Kafka broker.
+     * @param event the events to publish on the Kafka broker.
      */
-    protected void send(List<? extends EventMessage<?>> events) {
-        final Map<? super EventMessage<?>, MonitorCallback> monitorCallbacks = messageMonitor
-                .onMessagesIngested(events);
-        Producer<K, V> producer = producerFactory.createProducer();
-        ConfirmationMode cm = producerFactory.confirmationMode();
-        try {
-            if (cm.isTransactional()) {
-                tryBeginTxn(producer);
-            }
-            Map<Future<RecordMetadata>, ? super EventMessage<?>> publishStatuses = publishToKafka(events, producer);
-            if (CurrentUnitOfWork.isStarted()) {
-                handleActiveUnitOfWork(producer, publishStatuses, monitorCallbacks, cm);
-            } else if (cm.isTransactional()) {
-                tryCommit(producer, monitorCallbacks);
-            } else if (cm.isWaitForAck()) {
-                waitForPublishAck(publishStatuses, monitorCallbacks);
-            }
-        } finally {
-            if (!CurrentUnitOfWork.isStarted()) {
-                tryClose(producer);
-            }
-        }
-    }
-
-    /**
-     * Send's event messages to Kafka.
-     *
-     * @param events   list of event messages to publish.
-     * @param producer Kafka producer used for publishing.
-     * @return Map containing futures for each event that was published to kafka. You can interact with a specific
-     * {@link Future} to check whether a giSpringAxonAutoConfigurerTestven message was published successfully or not.
-     */
-    private Map<Future<RecordMetadata>, ? super EventMessage<?>> publishToKafka(List<? extends EventMessage<?>> events,
-                                                                                Producer<K, V> producer) {
-        Map<Future<RecordMetadata>, ? super EventMessage<?>> results = new HashMap<>();
-        events.forEach(event -> results.put(producer.send(messageConverter.createKafkaMessage(event, topic)), event));
-        return results;
-    }
+    public <T extends EventMessage<?>> void send(T event) {
```

Comment:
Github makes it tough to look at this change, but I think I get the gist.
Instead of starting this class through the auto-configuration, it's the `KafkaSendingEventHandler` which will call "this" `send(T)` method do publish a single event through the provided `Producer`.

Main differences:
 1. A single event is published instead of a list, which follows from the change in API of the former `StreamableMessageSource` approach in favor of the `@EventHandler` annotated method in the `KafkaSendingEventHandler`. This essential means you've adjusted this class from being a "intriguing" Event Processor implementation to a regular Event Handling Component.
 2. The assumption is made an active unit of work is always present, due to the `CurrentUnitOfWork.get()`. As such, the `handleActiveUnitOfWork` and it's afterCommit/rollback hooks could be replaced entirely for the new prepareCommit/rollback hooks.

Is my short summerization fair or am I missing important concepts from this Simon?

Reply:
Exactly. This rebuild simplified the vast amount of code and I in-sourced several methods.
In particular, the essence of the send method is now visible in the line
```
Future<RecordMetadata> publishStatus = producer.send(messageConverter.createKafkaMessage(event, topic));
``` 
The supporting `try-` methods are handling transaction preparation and completion before and after it.

Reply:
Sweet, good to hear I got it right! :-)